### ISSUE TITLE
A way to control client clipboard from server shell

### ISSUE THREAD

**Issue by j-xella** [wez/wezterm#3979]
**Is your feature request related to a problem? Please describe.**

When I connect from windows to headless linux box, without XServer, working with clipboard is difficult from the terminal/nvim. nvim allows clipboard integration through 3rd party utilities (`:he clipboard-tool`). It would be great if such utilities exist that would work purely with wezterm.

**Describe the solution you'd like**

A capability similar to `xclip` or `xsel`:

***Copying:***

1. In the shell: `echo "the text" | wezclip --copy`
2. wezclip sends the text somehow to wezterm-gui in which the shell is displayed
3. wezterm-gui sets the received text to the clipboard on the machine where zezterm-gui is running

Pasting:

1. In the shell: `wezclip --paste | consume ...`
2. wezclip sends a request to wezterm-gui in which the shell is displayed
3. wezterm-gui returns the contents of clipboard on the machine where zezterm-gui is running
4. wezclip dumps the received text to stdout

**Describe alternatives you've considered**

 * Ctrl+Shift+V can often be used in place of paste
 * WezTerm allows for screen copying, but it may be very inconvenient for bigger chunks of text in editors
 * Third-party utilities like lemonade

**Comment by wez** [comment]
wezterm supports the OSC 52 escape sequence that allows setting the clipboard.
wezterm does not support reading the clipboard via escape sequence, for your security and privacy protection.

https://gist.github.com/carloseberhardt/d9433f308e31c3ffebc34d07da074e22 is an example shell script that I just found with a quick internet search that shows how you might use OSC 52 to copy to the clipboard.

**Comment by j-xella** [comment]
True, reading clipboard poses a security issue. But then it could be disabled by default in wezterm, with possibility for the user to enable it, if they are sure what they are doing? Possibly on per-domain basis?

At the end of the day, if I were able to run X-Server on my Windows box, I would be exposed to the same clipboard-related security problems. If that is acceptable for me, allowing wezterm to read my clipboard would be acceptable as well.

**Comment by cbackas** [comment]
@wez Does this all apply to using the unix domain as well? nvim yanking to sys clipboard would need OSC52 and nvim pasting from your system clipboard is not possible?

I had a feeling my transition from tmux was going too smoothly and then i realized my clipboard workflow is all out of wack in wezterm :P

**Comment by dsully** [comment]
I believe I'm running into this issue with Neovim 0.10 (currently nightly) feature that supports OSC 52 pasting.

Neovim hangs when run on a remote server via SSH via WezTerm.

https://github.com/neovim/neovim/pull/25872

**Comment by NickJAllen** [comment]
It would be very useful to allow the user to customise this security setting rather than forcing it on all users and causing neovim to hang. It could have a default setting that is disallows read of clipboard (but even then I think returning an empty string rather than hanging the process would be better.) I could see 3 options: never allow, ask user, always allow.
